src/python/CRABInterface/Utils.py

```diff
@@ -86,35 +86,67 @@ def execute_command(command, logger, timeout):
 centralCfgFallback = None
 def getCentralConfig(extconfigurl, mode):
     """Utility to retrieve the central configuration to be used for dynamic variables
-
     arg str extconfigurl: the url pointing to the exteranl configuration parameter
     arg str mode: also known as the variant of the rest (prod, preprod, dev, private)
     return: the dictionary containing the external configuration for the selected mode."""
 
     global centralCfgFallback
-    hbuf = StringIO.StringIO()
-    bbuf = StringIO.StringIO()
-
-    curl = pycurl.Curl()
-    curl.setopt(pycurl.URL, extconfigurl)
-    curl.setopt(pycurl.WRITEFUNCTION, bbuf.write)
-    curl.setopt(pycurl.HEADERFUNCTION, hbuf.write)
-    curl.setopt(pycurl.FOLLOWLOCATION, 1)
-    curl.perform()
-    curl.close()
-
-    header = ResponseHeader(hbuf.getvalue())
-    if (header.status < 200 or header.status >= 300):
-        msg = "Reading %s returned %s." % (extconfigurl, header.status)
-        if centralCfgFallback:
-            msg += "\nUsing cached values for external configuration."
-            cherrypy.log(msg)
-            return centralCfgFallback
-        else:
-            cherrypy.log(msg)
-            raise ExecutionError("Internal issue when retrieving external confifuration from %s" % extconfigurl)
 
-    centralCfgFallback = json.loads(bbuf.getvalue())[mode]
+    def retrieveConfig(externalLink):
+
+        hbuf = StringIO.StringIO()
+        bbuf = StringIO.StringIO()
+
+        curl = pycurl.Curl()
+        curl.setopt(pycurl.URL, externalLink)
+        curl.setopt(pycurl.WRITEFUNCTION, bbuf.write)
+        curl.setopt(pycurl.HEADERFUNCTION, hbuf.write)
+        curl.setopt(pycurl.FOLLOWLOCATION, 1)
+        curl.perform()
+        curl.close()
+
+        header = ResponseHeader(hbuf.getvalue())
+        if (header.status < 200 or header.status >= 300):
+            msg = "Reading %s returned %s." % (externalLink, header.status)
+            if centralCfgFallback:
+                msg += "\nUsing cached values for external configuration."
+                cherrypy.log(msg)
+                return centralCfgFallback
+            else:
+                cherrypy.log(msg)
+                raise ExecutionError("Internal issue when retrieving external configuration from %s" % externalLink)
+        return bbuf.getvalue()
```

Comment:
I like to call functions in a separate statement then the one with return. Makes debugging easier in case of problem. I do not expect problems here, but simply as a general rule. Please do something like
```
jsonConfig = bbuf.getvalue()
return jsonConfig
```